### ISSUE TITLE
StrictSignal.map returns Signal instead of StrictSignal

### ISSUE THREAD

**Issue by ivan-klass** [raquo/Airstream#134]
```scala
val myVar = Var(42)


def myFunc(ss: StrictSignal[Int]) = ???

myFunc(myVar.signal) // ok
myFunc(myVar.signal.map(_ * 2)) // compile error


```

**Comment by raquo** [comment]
Yes, we'll have that eventually. We already have the mechanism to implement this, main issue is that I want to make other map-like operators like `mapSome` to work similarly, but that's a lot bigger task, as I need to figure out a new abstraction to implement that without massive repetition.

So I was planning to do that in 18.0, but if my schedule allows, I could add just the strict `map` to 17.2, and the rest later.

This was previously mentioned in https://github.com/raquo/Airstream/issues/119 but I guess it's good to have a separate issue now that zoomLazy itself is done.

**Comment by ivan-klass** [comment]
@raquo thank you. I've end up with the following work-around. Not sure it's fully safe but it works.

```scala
type Cleanup = Modifier[Base]
extension [A](ss: StrictSignal[A])
  def strictMapped[B](f: A => B): (StrictSignal[B], Cleanup) =
    val mVar = Var(f(ss.now()))
    val owner = new ManualOwner
    ss.map(f).addObserver(mVar.writer)(using owner)
    (mVar.signal, onUnmountCallback(_ => owner.killSubscriptions()))
```
used like
```scala

def myFunc(ss: StrictSignal[Int]): HtmlElement = ???

val myVar = Var(42)
val (strictMappedSignal, cleanUp) = myVar.signal.strictMapped(_ * 2)

div(
  myFunc(strictMappedSignal),
  cleanUp
)

```

**Comment by raquo** [comment]
If you want a workaround for now, just make your extension method
instantiate and return a LazyStrictSignal. See how Var.zoomLazy method does
it.

**Comment by raquo** [comment]
Implemented in 17.2.0 as `mapLazy`. Can't really change `map` return type of an open trait without a major release.

A future major version will expand this to more operators, but I can't really create an issue for that yet because I'm not yet sure how it should look technically. I plan to look into that in the coming months.
